rust/core-lib/src/editor.rs

```diff
@@ -345,22 +345,31 @@ impl Editor {
                 // backspace deletes max(1, tab_size) contiguous spaces
                 let (_, c) = view.offset_to_line_col(&self.text, region.start);
 
-                let tab_off = c & config.tab_size;
+                // Offset of cursor position from nearest tab
+                let tab_off = c % config.tab_size;
                 let tab_size = config.tab_size;
-                let tab_size = if tab_off == 0 { tab_size } else { tab_off };
+
+                // Size of tab, counted from cursor to nearest tab
+                let tab_size = if tab_off == 0 { if c > tab_off { tab_size } else { c - tab_off } } else { tab_off };
                 let tab_start = region.start.saturating_sub(tab_size);
-                let preceded_by_spaces = region.start > 0 &&
-                    (tab_start..region.start).all(|i| self.text.byte_at(i) == b' ');
+                let preceded_by_spaces = region.start > 0 && tab_size > 0
+                && self.text.slice_to_string(tab_start, region.start).chars().all(|i| i.is_whitespace());
```

Comment:
I'm not sure about this change; among other things it now means that if tab_size is 4 this will delete three tabs and a space on backspace. In addition it requires us to allocate a string, instead of doing the comparison in the existing buffer.